force-app/main/default/lwc/geFormRenderer/geFormRenderer.js

```diff
@@ -943,24 +971,34 @@ export default class GeFormRenderer extends LightningElement{
             return false;
         }
 
-        return gift.isAuthorized();
+        return gift.hasElevateRemovableStatus();
     }
 
     shouldNullPaymentRelatedFields() {
+        return this.shouldNullFormerCreditCardPayment() || this.shouldNullFormerAchPayment();
+    }
+
+    shouldNullFormerCreditCardPayment() {
         return (this.isGiftAuthorized() || this.isGiftExpired())
             && this.selectedPaymentMethod() !== PAYMENT_METHOD_CREDIT_CARD;
     }
 
+    shouldNullFormerAchPayment() {
+        return this.isGiftPending() && this.selectedPaymentMethod() !== PAYMENT_METHOD_ACH;
+    }
+
     async handleRemoveFromElevateBatch(tokenizedGift) {
         const gift = new Gift(this.giftInView);
         const result = {hasError: false, wasRemoved: false};
 
         try {
             if (await this.shouldRemoveFromElevateBatch(gift, !!tokenizedGift)) {
-                await this.currentElevateBatch.remove(gift.asDataImport());
+                await this.currentElevateBatch.remove(gift);
                 if (!tokenizedGift) { this.handleNullPaymentFieldsInFormState(); }
+
                 result.wasRemoved = true;
             }
+            this.nullGiftFieldsForTypeConversion();
```

Comment:
Should we be calling this function in the `if` clause above? is there a reason for us to null these payment fields if the gift should not be removed from the Elevate Batch?

Reply:
@voduyemi I was thinking we would also want to null these fields if a user switches gift types for non-Elevate gifts as well

Reply:
That makes sense